File: solutions/python/collatz-conjecture/collatz_conjecture.py
import itertools

def steps(number):
    if number <= 0:
        raise ValueError('Only positive integers are allowed')

    for step in itertools.count():
        if number == 1:
            return step
        
        if number % 2 == 1:
            number = number * 3 + 1
        else:
            number /= 2
        
    return None

File: solutions/csharp/collatz-conjecture/CollatzConjecture.cs
using System;

public static class CollatzConjecture
{
    public static int Steps(int number)
    {
        if (number < 1)
        {
            throw new ArgumentOutOfRangeException(nameof(number), number, "only positive integers are allowed");
        }
        
        int steps = 0;

        while (number > 1)
        {
            if (number % 2 == 0)
            {
                number /= 2;
            }
            else
            {
                number = number * 3 + 1;
            }
            steps += 1;
        }
        
        return steps;
    }
}